lib/Driver/ToolChains.h

```diff
@@ -96,12 +96,20 @@ class LLVM_LIBRARY_VISIBILITY GenericUnix : public ToolChain {
   ~GenericUnix() = default;
 };
 
+class LLVM_LIBRARY_VISIBILITY Android : public GenericUnix {
+protected:
+  std::string getTargetForLinker() const override;
+
+  bool shouldProvideRPathToLinker() const override;
+public:
+  Android(const Driver &D, const llvm::Triple &Triple) : GenericUnix(D, Triple) {}
+  ~Android() = default;
+};
+
 class LLVM_LIBRARY_VISIBILITY Cygwin : public GenericUnix {
 protected:
   std::string getDefaultLinker() const override;
 
-  bool shouldSpecifyTargetTripleToLinker() const override;
```

Comment:
Ah, we probably want to include the negative check on Cygwin. For that you'll need to run FileCheck again on the same input but with a different prefix, something like

```
// RUN: FileCheck -check-prefix CYGWIN-x86_64-NEGATIVE %s < %t.cygwin.txt

// CYGWIN-x86_64-NEGATIVE-NOT: -target
```